python/rmm/_lib/cuda_stream_view.pyx

```diff
@@ -26,7 +28,10 @@ cdef class CudaStreamView:
             CUDA stream to wrap, default 0
         """
         if (stream == 0):
-            self.c_obj.reset(new cuda_stream_view())
+            if int(os.environ.get("CUDA_PTDS", "0")) != 0:
+                self.c_obj.reset(new cuda_stream_view(<cudaStream_t>2))
```

Comment:
Expose `rmm::cuda_stream_per_thread` to Cython so you don't have to hard code it as a magic number.

Reply:
Actually are you sure this should be in the ctor? If you do this, there will be no way to use the default stream if they set the env variable.

Reply:
Yeah we might want to consider defining this at the module level. Maybe have a variable called `sv` or similar. Then we can use that directly here. WDYT?

Reply:
> Actually are you sure this should be in the ctor? If you do this, there will be no way to use the default stream if they set the env variable.

If we are forcing PTDS (through the environment variable), then we want to change that behavior. Otherwise, the user would simply not use the environment variable. What we want here is that everything magically becomes PTDS.

Reply:
> Yeah we might want to consider defining this at the module level. Maybe have a variable called sv or similar. Then we can use that directly here. WDYT?

I'm not sure what you're suggesting here, is that just about replacing the object we return by a pre-built one?

Reply:
I think this is a case similar to how we handle import time behavior based on environment variables as well where we can have a `RAPIDS_*` variable that all RAPIDS libraries look for as well as a library specific `RMM_*` variable.

Reply:
> I think this is a case similar to how we handle import time behavior based on environment variables as well where we can have a `RAPIDS_*` variable that all RAPIDS libraries look for as well as a library specific `RMM_*` variable.

That's fine with me too, but as per https://github.com/rapidsai/rmm/pull/633#issuecomment-731407114 I was just thinking of what would people feel with a variable that can be shared by other projects, instead of each having its own variable and forcing the user to potentially enable 3 or 4 of them at each time.

Reply:
Calling it `CUDA_*` implies that it's an official CUDA feature, which it is not.